source/Irrlicht/CMakeLists.txt

```diff
@@ -24,6 +24,18 @@ elseif(MSVC)
 	add_compile_options(/GR- /Zl)
 endif()
 
+# Enable SSE for floating point math on 32-bit x86 by default
+# reasoning see minetest issue #11810 and https://gcc.gnu.org/wiki/FloatingPointMath
+if(CMAKE_SIZEOF_VOID_P EQUAL 4)
+	enable_language(C)
+	include(CheckCSourceCompiles)
+	check_c_source_compiles("#ifndef __i686__\n#error\n#endif\nint main(){}" IS_I686)
+	if(IS_I686)
+		message(STATUS "Detected Intel x86: using SSE instead of x87 FPU")
+		add_compile_options(-mfpmath=sse -msse)
+	endif()
+endif()
```

Comment:
This needs to be inside the `if(CMAKE_CXX_COMPILER_ID ...`

You can add this inside the other branch just like Minetest:
<pre>
if(CMAKE_SIZEOF_VOID_P EQUAL 4)
    add_compile_options(/arch:SSE)
endif()
</pre>

Reply:
I moved the sse fix into the CMAKE_CXX_COMPILER_ID if block.

Also, does **/arch:SSE** need to be enabled on MSVC for a non-intel cpu (such as. amd)

Reply:
Like I said both AMD and Intel are affected because they implement the same standard.